src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIKnight.java

```diff
@@ -111,12 +111,12 @@ protected int getAttackDelay()
      */
     protected IAIState attackProtect()
     {
-        final int shieldSlot = InventoryUtils.findFirstSlotInItemHandlerWith(getInventory(),
-          Items.SHIELD);
+        final int shieldSlot = InventoryUtils.findFirstSlotInItemHandlerWith(getInventory(), Items.SHIELD);
 
         if (target != null && target.isAlive())
         {
-            if (shieldSlot != -1)
+            final UnlockResearchEffect effect = worker.getCitizenColonyHandler().getColony().getResearchEffects().getEffect("Shield Usage", UnlockResearchEffect.class);
```

Comment:
This is 6 classes further in the review and here i understood what UnlockResearchEffect does..... Not well named.... No comment explaining what this variable is doing and checking.

Reply:
BTW this can simply be a boolean check -> getResearchEffects.has("Shield Usage").

Reply:
Also Constants!